gsa/src/web/pages/extras/cvsscalculatorpage.js

```diff
@@ -359,21 +350,388 @@ const CvssCalculator = ({gmp, onInteraction, ...props}) => {
   );
 };
 
-CvssCalculator.propTypes = {
-  gmp: PropTypes.gmp.isRequired,
-  onInteraction: PropTypes.func.isRequired,
+const CvssV3Calculator = ({props}) => {
+  const [, renewSession] = useUserSessionTimeout();
+
+  const [state, setState] = useState({
+    attackVector: 'NETWORK',
+    attackComplexity: 'LOW',
+    privilegesRequired: 'NONE',
+    userInteraction: 'NONE',
+    scope: 'UNCHANGED',
+    authentication: 'NONE',
+    integrityImpact: 'NONE',
+    availabilityImpact: 'NONE',
+    cvssVector: 'CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N',
+    userVector: 'CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:N',
+    cvssScore: 0,
+  });
+
+  useEffect(() => {
+    const {location} = props;
+
+    if (
+      isDefined(location) &&
+      isDefined(location.query) &&
+      isDefined(location.query.cvssVector)
+    ) {
+      const {cvssVector} = location.query;
+      setState(vals => ({...vals, cvssVector, userVector: cvssVector}));
+      handleVectorChange();
+
+      let newScore = calculateV3Score({
+        attackVector,
+        attackComplexity,
+        privilegesRequired,
+        userInteraction,
+        scope,
+        confidentialityImpact,
+        integrityImpact,
+        availabilityImpact,
+      });
+      setState(vals => ({...vals, cvssScore: newScore}));
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    handleVectorChange();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [state.userVector]);
+
+  const calculateVector = newVector => {
+    const {
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+    } = state;
+
+    const cvssVector = parseCvssV3BaseVector({
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+      ...newVector,
+    });
+
+    setState(vals => ({
+      ...vals,
+      ...newVector,
+      cvssVector,
+      userVector: cvssVector,
+    }));
+
+    let newScore = calculateV3Score({
+      attackVector,
+      attackComplexity,
+      privilegesRequired,
+      userInteraction,
+      scope,
+      confidentialityImpact,
+      integrityImpact,
+      availabilityImpact,
+    });
+    setState(vals => ({...vals, cvssScore: newScore}));
+  };
+
+  const handleInteraction = () => {
+    renewSession();
+  };
```

Comment:
HandleInteraction can be replaced by calling renewSession directly. Or if you want to keep the name just do
```suggestion
  const handleInteraction = renewSession;
```